sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/runtime/MicroBatchExecution.scala

```diff
@@ -344,9 +344,40 @@ class MicroBatchExecution(
     setLatestExecutionContext(execCtx)
 
     populateStartOffsets(execCtx, sparkSessionForStream)
+
+    // SPARK-53941: This code path is executed for the first batch, regardless of whether it's a
+    // fresh new run or restart.
+    disableAQESupportInStatelessIfUnappropriated(sparkSessionForStream)
```

Comment:
Do you have any plans to try to either remove this or have a way to opt-out since other checks _should_ theoretically be enough when choosing wither to insert an adaptive plan? We (sadly) only recently discovered the performance benefits of manually re-enabling AQE inside a FEB and it would be nice to have that work by default, and I think this is still preventing that from happening?

Reply:
It doesn't not prevent you to do that unless your "streaming part" (before FEB) of the query is stateful. Do I miss something?

Reply:
That is very dependent to the DSv1 sink implementation - FEB sink is somehow safe because we finalize RDD plan for streaming part. If not, it should have been problematic likewise I mentioned in the PR description. I'd still leave this to user's risk - you can still override AQE config in FEB sink after this fix to get the same behavior (which I really hate that config is overridable in FEB sink but...)